### ISSUE TITLE
[RFC] Endpoint in plugins

### ISSUE THREAD

**Issue by David-Maisonave** [stashapp/stash#5103]
## Endpoint in plugins (auto populate)
I have a plugin ([RenameFile](https://github.com/David-Maisonave/Axter-Stash/tree/main/plugins/RenameFile)) which has a UI field called **GraphQL Endpoint** with a default value of (http://localhost:9999/graphql). I want to try to populated this field automatically and remove it from the UI.

I notice that sys.stdin.read() returns the following:
`{"server_connection":{"Scheme":"http","Host":"0.0.0.0","Port":9999`

Is it safe for my source code to convert "0.0.0.0" to **localhost** when setting **GraphQL Endpoint**?
I tried **http://0.0.0.0:9999**, but I got an error on the browser. So I'm thinking that if the value is "0.0.0.0", I'll have to change it to **localhost**. If the value is anything else, I should be able to use that value in the **GraphQL Endpoint**.
Example:
```
endpointHost = json_input['server_connection']['Host']
if endpointHost == "0.0.0.0":
    endpointHost = "localhost"
myEndpoint = f"{json_input['server_connection']['Scheme']}://{endpointHost}:{json_input['server_connection']['Port']}/graphql"
```
- **Does that look right?**
- Is there a better method to get the **GraphQL Endpoint**?
- Why is the Host set to "0.0.0.0" by default instead of "127.0.0.1"?

## stash.configure_plugin (Does it work)
Every time I call **stash.configure_plugin**, the code hangs. Is this function still working?
I tried using it exactly as it's used in another plugin.
- Original plugin code:
  - `stash.configure_plugin("stashdb-performer-gallery", {"queue": ""})`
- With my version of the code:
  - `stash.configure_plugin("renamefile", {"zgraphqlEndpoint": "http://localhost:9999/graphql"})`

I added exception handling, and I got the following error message:
`Error: 422 Unprocessable Entity query failed. v0.26.2-2`
Here's what I got on the tracestack:
```
Traceback (most recent call last):
  File "C:\Users\admin3\.stash\plugins\RenameFile\renamefile.py", line 110, in <module>
    stash.configure_plugin("renamefile", {"zgraphqlEndpoint": "http://localhost:9999/graphql"})
  File "C:\Users\david\AppData\Local\Programs\Python\Python310\lib\site-packages\stashapi\stashapp.py", line 375, in configure_plugin
    plugin_values = self.find_plugin_config(plugin_id)
  File "C:\Users\david\AppData\Local\Programs\Python\Python310\lib\site-packages\stashapi\stashapp.py", line 392, in find_plugin_config
    return self.find_plugins_config(plugin_id)
  File "C:\Users\david\AppData\Local\Programs\Python\Python310\lib\site-packages\stashapi\stashapp.py", line 405, in find_plugins_config
    config = self.call_GQL(query, {"input": plugin_ids})["configuration"]["plugins"]
  File "C:\Users\david\AppData\Local\Programs\Python\Python310\lib\site-packages\stashapi\stashapp.py", line 216, in call_GQL
    return self._GQL(query, variables)
  File "C:\Users\david\AppData\Local\Programs\Python\Python310\lib\site-packages\stashapi\classes.py", line 227, in _GQL
    return self._handle_GQL_response(response)
  File "C:\Users\david\AppData\Local\Programs\Python\Python310\lib\site-packages\stashapi\classes.py", line 270, in _handle_GQL_response
    raise Exception(error_msg)

```

I'm not sure what that means. Is **stash.configure_plugin** broken, or am I using it the wrong way?

**Comment by stg-annon** [comment]
accessing this info from the values passed in `server_connection` is the intended way to connect for plugins and is essentially the only way for them to automatically connect to a stash with a password set, since you are using `stashapp-tools` all you would need to do is pass it the server connection value and it would handle the rest

```py

from stashapi.stashapp import StashInterface

json_input = json.loads(sys.stdin.read())

stash = StashInterface(json_input["server_connection"])
```

for initializing config values you can use `find_plugin_config()` with the plugin ID and the default values you want to set, it your case it would be something like

```py
default_settings = {
    "performerAppend": False,
    "studioAppend": False,
    "tagAppend": False,
    "z_keyFIeldsIncludeInFileName": False,
    "zafileRenameViaMove": False,
    "zfieldKeyList": DEFAULT_FIELD_KEY_LIST,
    "zgraphqlEndpoint": DEFAULT_ENDPOINT,
    "zmaximumTagKeys": 12,
    "zpathToExclude": "",
    "zseparators": DEFAULT_SEPERATOR,
    "ztagWhitelist": "",
    "zzdebugTracing": False,
    "zzdryRun": False,
}
plugin_configuration = stash.find_plugin_config("renamefile", default_settings)
```

I suspect the one of the config functions is returning `None` which may be causing this issue as it would be expecting a empty dict, not sure if this is from a change in how stash returns from the API or a latent bug in `stashapp-tools`, in either case I will address it within the library

**Comment by stg-annon** [comment]
published under [v0.2.47](https://github.com/stg-annon/stashapp-tools/releases/tag/v0.2.47)

also for future reference this would not be an issue for stash but for the [stashapp-tools](https://github.com/stg-annon/stashapi) repo

**Comment by David-Maisonave** [comment]
> also for future reference this would not be an issue for stash but for the [stashapp-tools](https://github.com/stg-annon/stashapi) repo

Thank you.  I will continue this discussion there.
